core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2187,7 +2187,8 @@ private[spark] class DAGScheduler(
                   log"we will roll back and rerun below stages which include itself and all its " +
                   log"indeterminate child stages: ${MDC(STAGES, rollingBackStages)}")
               }
-
+              failedStage.markResubmitInFetchFailed()
+              mapStage.markResubmitInFetchFailed()
```

Comment:
nit: `markResubmitInFetchFailed` -> `setResubmitByFetchFailure`.

Also, please reset this to `false` - when stage transitions to running - so that scheduler state is consistent.
`submitMissingTasks` preferably, or in `submitStage`

Reply:
Done,  this can make the variable meaning more accurate. 
But in addition, when `submittingMissingTasks`,  the stage will be added to the `runningStages`, we can always cancel job normally.